packages/nx/src/daemon/server/sync-generators.ts

```diff
@@ -136,8 +141,16 @@ export function collectAndScheduleSyncGenerators(
       return;
     }
 
-    const { projects } =
-      readProjectsConfigurationFromProjectGraph(projectGraph);
+    // Fetch the latest graph instead of using the captured one which might be stale
+    const { projectGraph: latestGraph, error } =
```

Comment:
The `projectGraph` has been used plenty in this function... wouldn't those also be outdated? What is sending this function an outdated graph?